enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -199,214 +153,90 @@ rules:
+
+- reads the `Share` object referenced in the `volumeMetadata` which is available in the `NodePublishVolumeRequest` that comes into `NodePublishVolume`.
+- the name, namespace, and `ServiceAccount` of the Pod wanting the volume is also included by the kubelet in the `NodePublishVolumeRequest` under the keys `csi.storage.k8s.io/pod.name`, `csi,storage.k8s.io/pod.namespace`, and `csi.storage.k8s.io/serviceAccount.name` in the map returned by the `GetVolumeContext()` call.
 - the `podInfoOnMount` field of the CSI Driver spec triggers the setting of these values; here is
 [an example from the hostpath plugin](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.17/hostpath/csi-hostpath-driverinfo.yaml#L12)
-- with all this necessary metadata, the CSI driver then
-- reads of the `Secret` or `ConfigMap` noted in the `ProjectedResource`, storing it on disk on initial
-reference to said `Secret` or `ConfigMap`
-- creation of a tmpfs filesystem for each requesting `ServiceAcount` that contains the `Secret` / `ConfigMap` data
-- creation of k8s watches `ProjectedResources`, and for `Secret` / `ConfigMaps` in the namespaces listed in each discovered
-`ProjectedResources`
-- final returning of a volume that is based off of the `ServiceAccount` specific tmpfs for the data noted in the
-`volumeAttributes`
 
-After initial set up of things, steady state CSI driver activities include:
-- as updates of the watched `Secrets` and `ConfigMaps` come in, SAR checks will be done for each of the previous requesting
-`ServiceAccounts` being tracked against the updated `Secret` and `ConfigMap`
-- for any `ServiceAccounts` that no longer have access, their tmpfs filesystems are unmounted, effectively removing their access to
-the data (though in prototyping and implementation, we'll need focused testing that this scales and mount propagation
-does not become a gating factor)
-- the SAR checks and if need be removal of access to data can facilitate cluster admin scenarios noted in the `User Stories`
-section like credential rotation.
-- But for `ServiceAccounts` that pass the SAR check, the tmpfs file with the data for the `Secret` / `ConfigMap` is updated
+With all this necessary metadata, the CSI driver then:
 
+- Performs a subject access review (SAR) check to verify the pod's service account has permission to GET the referenced Share resource.
+- If the SAR check succeeds, reads of the `Secret` or `ConfigMap` noted in the `Share`, storing it on disk on initial reference to said `Secret` or `ConfigMap`.
+- Creates a tmpfs filesystem for each requesting `ServiceAcount` that contains the `Secret` / `ConfigMap` data.
+- Creates k8s watches `Shares`, and for `Secret` / `ConfigMaps` in the namespaces listed in each discovered `Share`.
+- Returns a volume that is based off of the `ServiceAccount` specific tmpfs for the data noted in the `volumeAttributes`.
 
-#### An overview of CSI and the main K8S hook points
```

Comment:
Moved this section to an appendix with further details

Reply:
SGTM